### PR TITLE
refactor(account-lib): add `getPublicKey()`/`getPrivateKey()`

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Reduce copypasta and low-level utxolib access:

* add method `getPublicKey({ compressed: boolean })`
* add method `getPrivateKey()` methods
* assume `keyPair` property is always set (like the type implies), and
  that it always set to `this.hdNode.keyPair`.

Issue: BG-34381

### CODE REVIEW

**Review by sangaman — DISMISSED** [review]
Changes look good. It's not clear to me whether `getPublicKey`/`getPrivateKey` will give the same values as the old `HDNode.fromBase58...` code. But it looks like there's already quite a bit of existing testing around `getKeys` that would catch any deviations.

File: modules/account-lib/src/coin/stx/keyPair.ts [hunk 1]
```diff
@@ -72,11 +73,9 @@ export class KeyPair extends Secp256k1ExtendedKeyPair {
     }
 
     if (isValidXpub(pub)) {
-      this.hdNode = HDNode.fromBase58(pub);
-      this.hdNode.keyPair.compressed = false;
```
Comment (OttoAllmendinger) [inline modules/account-lib/src/coin/stx/keyPair.ts:80]: the parameter `compressed` in `getKeys` has precedence

File: modules/account-lib/src/coin/baseCoin/secp256k1ExtendedKeyPair.ts [hunk 2]
```diff
@@ -24,17 +33,25 @@ export abstract class Secp256k1ExtendedKeyPair implements BaseKeyPair {
     this.source = source;
   }
 
+  getPublicKey({ compressed }: { compressed: boolean }): Buffer {
+    return ECPair.fromPublicKey(this.keyPair.publicKey, { compressed }).publicKey;
+  }
+
+  getPrivateKey(): Buffer | undefined {
+    return this.keyPair.privateKey;
+  }
```
Comment (argjv) [inline modules/account-lib/src/coin/baseCoin/secp256k1ExtendedKeyPair.ts:42]: Add some JS docs explaining the format these keys are returned in.

Comment (OttoAllmendinger) [reply modules/account-lib/src/coin/baseCoin/secp256k1ExtendedKeyPair.ts]: done

**Comment by OttoAllmendinger** [comment]
> 774	ERROR in bitgo-account-lib.min.js from Terser
> 775	Unexpected token: punc ()) [bitgo-account-lib.min.js:11377,2]

I can't reproduce this locally :(

**Comment by OttoAllmendinger** [comment]
resolved compilation error

File: modules/account-lib/src/coin/avaxc/keyPair.ts [hunk 3]
```diff
@@ -20,9 +20,9 @@ export class KeyPair extends Secp256k1ExtendedKeyPair {
     super(source);
```
Comment (argjv) [inline modules/account-lib/src/coin/avaxc/keyPair.ts:20]: Shouldn't this `if` go into the base class now that it looks the same for almost all coins?

Comment (sangaman) [reply modules/account-lib/src/coin/avaxc/keyPair.ts:20]: Looks like xrp doesn't have the `if` statement and `eth` has one that's a bit different. Still maybe this logic could be moved into a member function called in the constructor which the two coins above override.

**Review by sangaman — APPROVED** [review]
Looks good but there might be even more copypasta to remove per jorge's comment if you think it's worthwhile.

File: modules/account-lib/src/coin/avaxc/keyPair.ts (hunk 3 above)
Comment (OttoAllmendinger) [reply modules/account-lib/src/coin/avaxc/keyPair.ts:20]: there is room for a lot more cleanup, but this PR was more about getting rid of utxolib

**Pull request merged**